packages/run-protocol/src/vpool-xyk-amm/addPool.js

```diff
@@ -12,50 +13,92 @@ const { details: X } = assert;
  * @param {ZCF} zcf
  * @param {(Brand) => boolean} isInSecondaries
  * @param {WeakStore<Brand,ZCFMint>} brandToLiquidityMint
+ * @param {WeakStore<Brand,ERef<Issuer>>} brandToLiquidityIssuer
  * @param {() => (secondaryBrand: Brand) => Promise<void>} getAddIssuerToReserve
  */
 export const makeAddIssuer = (
   zcf,
   isInSecondaries,
   brandToLiquidityMint,
+  brandToLiquidityIssuer,
   getAddIssuerToReserve,
 ) => {
   /**
+   * Add a new issuer. If we previously received a request for the same issuer,
+   * we'll return a promise or the issuer.
+   *
    * @param {Issuer} secondaryIssuer
    * @param {string} keyword
    */
   return async (secondaryIssuer, keyword) => {
-    const [secondaryAssetKind, secondaryBrand] = await Promise.all([
-      E(secondaryIssuer).getAssetKind(),
-      E(secondaryIssuer).getBrand(),
-    ]);
-
-    assert(
-      !isInSecondaries(secondaryBrand),
-      X`issuer ${secondaryIssuer} already has a pool`,
-    );
-    assert(
-      secondaryAssetKind === AssetKind.NAT,
-      X`${keyword} asset not fungible (must use NAT math)`,
-    );
-    const liquidityKeyword = `${keyword}Liquidity`;
-    zcf.assertUniqueKeyword(liquidityKeyword);
-
-    const mint = await zcf.makeZCFMint(
-      liquidityKeyword,
-      AssetKind.NAT,
-      harden({ decimalPlaces: 6 }),
+    return E.when(
```

Comment:
```javascript
export const makeAddIssuer = (
  zcf,
  isInSecondaries,
  brandToLiquidityMint,
  brandToLiquidityIssuer,
  getAddIssuerToReserve,
) => {
  /**
   * Add a new issuer. If we previously received a request for the same issuer,
   * we'll return a promise or the issuer.
   *
   * @param {Issuer} secondaryIssuer
   * @param {string} keyword
   */
  return async (secondaryIssuer, keyword) => {
    const [secondaryAssetKind, secondaryBrand] = await Promise.all([
      E(secondaryIssuer).getAssetKind(),
      E(secondaryIssuer).getBrand(),
    ]);

    assert(
      secondaryAssetKind === AssetKind.NAT,
      X`${keyword} asset not fungible (must use NAT math)`,
    );

    if (brandToLiquidityIssuer.has(secondaryBrand)) {
      return brandToLiquidityIssuer.get(secondaryBrand);
    }

    assert(
      !isInSecondaries(secondaryBrand),
      X`issuer ${secondaryIssuer} already has a pool`,
    );

    /** @template T @typedef {import('@endo/promise-kit').PromiseKit<T>} PromiseKit */
    /** @type {PromiseKit<Issuer<'nat'>>} */
    const promiseKit = makePromiseKit();
    brandToLiquidityIssuer.init(secondaryBrand, promiseKit.promise);

    const liquidityKeyword = `${keyword}Liquidity`;
    zcf.assertUniqueKeyword(liquidityKeyword);

    const { issuer } = await zcf.saveIssuer(secondaryIssuer, keyword);
    const mint = await zcf.makeZCFMint(
      liquidityKeyword,
      AssetKind.NAT,
      harden({ decimalPlaces: 6 }),
    );
    console.log(
      X`Saved issuer ${secondaryIssuer} to keyword ${keyword} and got back ${issuer}`,
    );
    // this ensures that getSecondaryIssuer(thisIssuer) will return even before the pool is created
    brandToLiquidityMint.init(secondaryBrand, mint);
    const { issuer: liquidityIssuer } = mint.getIssuerRecord();
    // defer lookup until necessary. more aligned with governed param we expect this to be eventually.
    const addIssuerToReserve = getAddIssuerToReserve();
    // tell the reserve about this brand, which it will validate by calling back
    // to AMM for the issuer
    await addIssuerToReserve(secondaryBrand);
    promiseKit.resolve(liquidityIssuer);
    brandToLiquidityIssuer.delete(secondaryBrand);
    return liquidityIssuer;
  };
};
```

Reply:
@erights   Your comment seems to prefer the form without `await`.   Which preference is stronger? @dtribble?

> Wonderful that you removed non-test awaits! Should really help us think about what happens during which turn.

Reply:
The problem was that the interlock was spread across turns. To me, this makes it more obvious and straightforward to prevent that

Reply:
BTW the "6" in there needs to be promoted to a constant